### DISCUSSION TITLE
Integrate lockFileMaintenance into the terragrunt manager

### DISCUSSION THREAD

**Discussion by norman-zon** [renovatebot/renovate#22618] [Suggest an Idea]
### Tell us more.

I would like to express my interest to have the functionality implemented in https://github.com/renovatebot/renovate/pull/20833 added into the terraform manager.

My planned workflow looks like this:

1. When adding a new` terragrunt.hcl `to a repo I run `terragrunt init -upgrade` locally once.
2. renovate runs provider updates on a infrequent schedule (once a month would be fine) and also updates the lockfiles along the providers
2. The renovate PRs are tested with Atlantis
3. If Atlantis shows no diffs I merge the PRs

The current implementation fails at point 3. because Atlantis is run with `terragrunt init` (I would like to avoid to run it with `init -upgrade` in case someone accidentally removes the version pinning).

So having the lockfile updates integrated with the provider upgrades would be the preferred and IMHO most clean solution here.
Otherwise I would have to merge the renovate PRs without being able to have them automatically tested and will see breakage only after the seperate `lockFileMaintenance` is run.

**Comment by HonkingGoose** [comment]
### Lockfile support for terragrunt or terraform?

> I would like to express my interest to have the functionality implemented in #20833 added into the terraform manager.

Do you mean terragrunt manager here?

### Background info

We recently merged this PR:

- #20833

That merge closed this issue:

- #13393

@viceice re-opened that issue because:

> reopen because we currently only support lockfile maintenance.

### Maintainers decide

I'll let the maintainers decide what to do here.

**Reply by norman-zon** [reply]
Yes, I mean the terragrunt manager, sorry.

I did not see https://github.com/renovatebot/renovate/issues/13393. I'll subscribe to the issue to follow further discussion.
